lib/ssz_ex.ex

```diff
@@ -527,45 +578,89 @@ defmodule LambdaEthereumConsensus.SszEx do
   defp decode_container(binary, module) do
     schemas = module.schema()
     fixed_length = get_fixed_length(schemas)
-    <<fixed_binary::binary-size(fixed_length), variable_binary::bitstring>> = binary
 
-    with {:ok, fixed_parts, offsets} <- decode_fixed_section(fixed_binary, schemas, fixed_length),
-         {:ok, variable_parts} <- decode_variable_section(variable_binary, offsets) do
-      {:ok, struct!(module, fixed_parts ++ variable_parts)}
+    if variable_size?(module) do
```

Comment:
Maybe we can benefit of having variable_size? be a guard and use it to pattern match in the decode_container definition? e.g.

```elixir
defp decode_container(binary, module) when variable_size?(module) do
 #...
end
```

If that's not possible we can do the same we did for vectors and lists, and have the if + `decode_variable_container` or `decode_fixed_container`